sys/net/gnrc/pktbuf/include/pktbuf_internal.h

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (C) 2014 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *               2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_gnrc_pktbuf
+ * @brief       Internal stuff for gnrc_pktbuf
+ *
+ * @{
+ *
+ * @file
+ * @brief   Interface definition for the global network buffer. Network devices
+ *          and layers can allocate space for packets here.
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ * @author  Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+#ifndef PKTBUF_INTERNAL_H
+#define PKTBUF_INTERNAL_H
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+#include "mutex.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Mutex used to provide mutually exclusive access to packet buffer
+ *          data structures
+ *
+ * @warning This mutex will be ***internally*** locked and unlocked and should
+ *          not be touched by external code
+ */
+extern mutex_t gnrc_pktbuf_mutex;
+
+#if IS_USED(MODULE_GNRC_PKTBUF_STATIC) || DOXYGEN
+/**
+ * @brief   The actual static buffer used when module gnrc_pktbuf_static is used
+ *
+ * @warning This is an internal buffer and should not be touched by external code
+ */
+extern uint8_t *gnrc_pktbuf_static_buf;
+#endif
+
+/**
+ * @brief   Check if the given pointer is indeed part of the packet buffer
+ *
+ * @warning This function is ***internal***.
+ *
+ * @param   ptr     pointer to check
+ *
+ * @retval  true    It is plausible, that this pointer belongs to data in
+ *                  the packet buffer
+ * @retval  false   @p ptr does not point to data in the packet buffer
+ */
+static inline bool gnrc_pktbuf_contains(void *ptr)
+{
+#if IS_USED(MODULE_GNRC_PKTBUF_STATIC)
+    return (unsigned)((uint8_t *)ptr - gnrc_pktbuf_static_buf) < CONFIG_GNRC_PKTBUF_SIZE;
+#else
+    (void)ptr;
+    return true;
+#endif
+}
+
+#if IS_USED(MODULE_GNRC_PKTBUF_STATIC) || DOXYGEN
+/**
+ * @brief   Release an internal buffer
+ *
+ * @warning This function is ***internal***. Use @ref gnrc_pktbuf_release to
+ *          release a packet buffer
+ *
+ * @param   data    pointer to the internal buffer to release
+ * @param   size    size of @p data in bytes
+ */
+void gnrc_pktbuf_free_internal(void *data, size_t size);
+#else
+static inline void gnrc_pktbuf_free_internal(void *data, size_t size)
+{
+    (void)size;
+    free(data);
+}
+#endif
```

Comment:
> There are some more functions for which you can do this ;-).

Are you referring to this? This would only work, if the function would be `inline`able in both cases. But in case of the static version, I think it is better without `inline`.

Reply:
Ok, I think I mis-looked then.